### PR TITLE
Fix typo in parameter name in cpp.py

### LINKED ISSUE #159239: [Inductor] Inconsistent results between eager execution and torch.compile
### 🐛 Describe the bug

When using torch.compile(..., backend="inductor") in the Qwen3 transformer model, a correctness discrepancy is observed in the RoPE computation. Below is a minimal reproducible example:
```python
import torch
import unittest
from typing import List


def rotate_half(x):
    """Rotates half the hidden dims of the input."""
    x1 = x[..., : x.shape[-1] // 2]
    x2 = x[..., x.shape[-1] // 2 :]
    return torch.cat((-x2, x1), dim=-1)

def apply_rotary_pos_emb(
    q: torch.Tensor,
    k: torch.Tensor,
    cos: torch.Tensor,
    sin: torch.Tensor,
    position_ids: torch.Tensor = None,
    unsqueeze_dim: int = 1,
) -> List[torch.Tensor]:
    """Applies Rotary Position Embedding to the query and key tensors.

    Args:
        q (`torch.Tensor`): The query tensor.
        k (`torch.Tensor`): The key tensor.
        cos (`torch.Tensor`): The cosine part of the rotary embedding.
        sin (`torch.Tensor`): The sine part of the rotary embedding.
        position_ids (`torch.Tensor`, *optional*):
            Deprecated and unused.
        unsqueeze_dim (`int`, *optional*, defaults to 1):
            The 'unsqueeze_dim' argument specifies the dimension along which to unsqueeze cos[position_ids] and
            sin[position_ids] so that they can be properly broadcasted to the dimensions of q and k. For example, note
            that cos[position_ids] and sin[position_ids] have the shape [batch_size, seq_len, head_dim]. Then, if q and
            k have the shape [batch_size, heads, seq_len, head_dim], then setting unsqueeze_dim=1 makes
            cos[position_ids] and sin[position_ids] broadcastable to the shapes of q and k. Similarly, if q and k have
            the shape [batch_size,s seq_len, heads, head_dim], then set unsqueeze_dim=2.
    Returns:
        `tuple(torch.Tensor)` comprising of the query and key tensors rotated using the Rotary Position Embedding.
    """
    cos = cos.unsqueeze(unsqueeze_dim)
    sin = sin.unsqueeze(unsqueeze_dim)
    q_embed = (q * cos) + (rotate_half(q) * sin)
    k_embed = (k * cos) + (rotate_half(k) * sin)
    return q_embed, k_embed


class TestQwen3MLP(unittest.TestCase):
    def test_rope(self):
        q = torch.randn([8, 32, 241, 128], dtype=torch.bfloat16)
        k = torch.randn([8, 8, 241, 128], dtype=torch.bfloat16)
        cos = torch.randn([8, 241, 128], dtype=torch.bfloat16)
        sin = torch.randn([8, 241, 128], dtype=torch.bfloat16)

        golden_q_embed, golden_k_embed = apply_rotary_pos_emb(q, k, cos, sin)

        compiled_func = torch.compile(apply_rotary_pos_emb, backend='inductor', fullgraph=True)
        test_q_embed, test_k_embed = compiled_func(q, k, cos, sin)

        assert torch.allclose(golden_q_embed, test_q_embed), f"[DISMATCH] golden_q_embed: {golden_q_embed}, test_q_embed: {test_q_embed}"
        assert torch.allclose(golden_k_embed, test_k_embed), f"[DISMATCH] golden_q_embed: {golden_k_embed}, test_q_embed: {test_k_embed}"


if __name__ == '__main__':
    unittest.main()
```

### Versions

PyTorch version: 2.6.0+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Debian GNU/Linux 12 (bookworm) (x86_64)
GCC version: (Debian 12.2.0-14) 12.2.0
Clang version: Could not collect
CMake version: version 3.25.1
Libc version: glibc-2.36

Python version: 3.11.2 (main, May  2 2024, 11:59:08) [GCC 12.2.0] (64-bit runtime)
Python platform: Linux-5.4.143.bsk.7-amd64-x86_64-with-glibc2.36
Is CUDA available: True
CUDA runtime version: 12.4.131
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA A800-SXM4-80GB
Nvidia driver version: 535.161.08
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.9.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv.so.9.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn.so.9.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_engines_precompiled.so.9.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_engines_runtime_compiled.so.9.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_graph.so.9.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_heuristic.so.9.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops.so.9.4.0
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   46 bits physical, 57 bits virtual
Byte Order:                      Little Endian
CPU(s):                          120
On-line CPU(s) list:             0-119
Vendor ID:                       GenuineIntel
Model name:                      Intel(R) Xeon(R) Platinum 8336C CPU @ 2.30GHz
CPU family:                      6
Model:                           106
Thread(s) per core:              2
Core(s) per socket:              30
Socket(s):                       2
Stepping:                        6
BogoMIPS:                        4589.21
Hypervisor vendor:               KVM
Virtualization type:             full
L1d cache:                       2.8 MiB (60 instances)
L1i cache:                       1.9 MiB (60 instances)
L2 cache:                        75 MiB (60 instances)
L3 cache:                        108 MiB (2 instances)
NUMA node(s):                    2
NUMA node0 CPU(s):               0-59
NUMA node1 CPU(s):               60-119
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.26.4
[pip3] nvidia-cublas-cu12==12.4.5.8
[pip3] nvidia-cuda-cupti-cu12==12.4.127
[pip3] nvidia-cuda-nvrtc-cu12==12.4.127
[pip3] nvidia-cuda-runtime-cu12==12.4.127
[pip3] nvidia-cudnn-cu12==9.1.0.70
[pip3] nvidia-cufft-cu12==11.2.1.3
[pip3] nvidia-curand-cu12==10.3.5.147
[pip3] nvidia-cusolver-cu12==11.6.1.9
[pip3] nvidia-cusparse-cu12==12.3.1.170
[pip3] nvidia-cusparselt-cu12==0.6.2
[pip3] nvidia-nccl-cu12==2.21.5
[pip3] nvidia-nvjitlink-cu12==12.4.127
[pip3] nvidia-nvtx-cu12==12.4.127
[pip3] torch==2.6.0
[pip3] triton==3.2.0
[conda] Could not collect

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @muchulee8 @amjames @aakhundov @coconutruben

### PR DESCRIPTION
**Pull request by Yuxingwang-intel**
Fixes #159239 
This PR fixes a typographical error in "torch/inductor/cpp.py", changing the parameter name from "use_compute_dtypes" to "use_compute_types".
The typo previously caused a runtime error when setting the TORCHINDUCTOR_EMULATE_PRECISION_CASTS=1. This fix ensures the parameter name is correct
This PR includes a unit test to ensure the corrected parameter works properly and prevents regression.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

### CODE REVIEW

**Comment by janeyx99** [comment]
This PR doesn't seem to address the linked issue...adding other reviewers to confirm

**Review by eellison — APPROVED** [review]

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -15456,7 +15456,14 @@ def fn(pytype, dtype):
                         ret_opt = fn_opt(pytype, dtype)
 
                 self.assertEqual(ret_opt, fn(pytype, dtype))
-
+def test_to_dtype_use_compute_types():
+    config.emulate_precision_casts=True
```
Comment (desertfire) [inline test/inductor/test_torchinductor.py]: Don't set this flag globally. Use `@config.patch` instead.

File: test/inductor/test_torchinductor.py [hunk 2]
```diff
@@ -15456,7 +15456,14 @@ def fn(pytype, dtype):
                         ret_opt = fn_opt(pytype, dtype)
 
                 self.assertEqual(ret_opt, fn(pytype, dtype))
-
+def test_to_dtype_use_compute_types():
```
Comment (desertfire) [inline test/inductor/test_torchinductor.py]: The indent looks weird. Is this a CPU-only test or not? Is it a part of `TestFull`?

File: test/inductor/test_torchinductor.py [hunk 3]
```diff
@@ -15456,7 +15456,14 @@ def fn(pytype, dtype):
...
+    config.emulate_precision_casts=True
+    torch.manual_seed(0)
```
Comment (desertfire) [inline test/inductor/test_torchinductor.py]: This shouldn't be needed if you move this test to `CommonTemplate` and use `self.common`.

**Review by desertfire — CHANGES_REQUESTED** [review]

**Comment by iremyux** [comment]
Adding `ciflow/win-arm64` label to trigger Windows Arm64 CI.
(This is for testing purposes only and isn’t related to any changes in this PR.)
